server/src/main/java/org/opensearch/extensions/rest/ExtensionRestRequest.java

```diff
@@ -118,6 +119,7 @@ public ExtensionRestRequest(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
+        validateRequestIssuerIdentity();
```

Comment:
This is not needed I think (the `out.writeString` will fail anyway), better have a constructor check:
```
        this.principalIdentifierToken = requireNotNull(principalIdentifie)r;

```